MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Xna.Framework
+{
+    public static partial class MathHelper
+    {
+        /// <summary>
+        /// Class for generating random numbers consistantly when using Mono or Microsoft compilers
+        /// </summary>
+        public class Random
+        {
+            #region Static methods
+
+            [ClsCompliant(false)]
+            public static ulong GenerateSeed()
+            {
+                byte[] guid = new Guid().ToByteArray();
+                return (BitConverter.ToUInt64(guid, 0) ^ BitConverter.ToUInt64(guid, 7));
+            }
+
+            #endregion
+
+            #region Private Fields
+
+            private ulong _seed;
+            private ulong _lastValue;
+
+            #endregion
+
+            #region Properties
+
+            [ClsCompliant(false)]
+            public ulong Seed
+            {
+                get
+                {
+                    return this._seed;
+                }
+                set
+                {
+                    this._seed = value;
+                    this._lastValue = value;
```

Comment:
I wouldn't have a problem with having both like a `Random.Seed` (being the starting seed) and `Random.State` (being the seed for the next random generation).  @KonajuGames ?